### PR TITLE
Fixed Command Injection vulnerability for android app command

### PR DESCRIPTION
**Pull request by MirceaConstantin**

### CODE REVIEW

**Comment by 69** [comment]
Still vulnerable with the following payload:
```bash
logkitty android app $'test\' "$(touch HACKED)" \''
```

File: src/android/adb.ts [hunk 1]
```diff
@@ -50,9 +50,10 @@ export function getApplicationPid(
   adbPath?: string
 ): number {
   let output: Buffer | undefined;
+  if (applicationId.match(/[;&|]/)) throw new CodeError(ERR_ANDROID_CANNOT_GET_APP_PID, ERR_ANDROID_UNPROCESSABLE_PID)
```
Comment (toufik-airane) [inline src/android/adb.ts:53]: The regexp doesn't fully mitigate the potential command injection.

**Review by toufik-airane — CHANGES_REQUESTED** [review]
Dear @MirceaConstantin,

The fix doesn't fully mitigate the potential command injection.
Thank you for your contribution.

**Comment by huntr-helper** [comment]
Sorry @MirceaConstantin, we enjoyed reviewing your fix but it has not been selected this time. If this bounty has not been closed, please feel free to try again with a new pull request! We appreciate your effort and look forward to reviewing more of your fixes in the future! :hammer: :sunglasses:

**Pull request closed without merging**